android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,360 @@
+					int commaIndex = 0;
+					for (KrollDict dict: daysOfTheWeek) {
+						weeklyRecurrencesString.append(dict.get(dayOfWeekKey).toString());
+						if (commaIndex < daysOfTheWeek.length) {
+							weeklyRecurrencesString.append(",");
+						}
+					}
+					finalRRule.append(weeklyRecurrencesString.toString());
+					finalRRule.append(";");
+					break;
+				// Case for monthly recurring events.
+				case CalendarModule.RECURRENCEFREQUENCY_MONTHLY:
+					StringBuilder monthlyReccurencesString = new StringBuilder("BYDAY=");
+					// daysOfTheWeek dictionary is with highest priority.
+					if (daysOfTheWeek.length > 0) {
+						monthlyReccurencesString.append(daysOfTheWeek[0].getInt(weekNumberKey));
+						// Potentially add week start (Sunday or Monday) different from the default one.
+						monthlyReccurencesString.append(weekdaysMap.keySet().toArray()[daysOfTheWeek[0].getInt(weekNumberKey)]);
+					} else {
+						// Case in which we do not have items in daysOfTheWeek array.
+						monthlyReccurencesString.append(String.valueOf(daysOfTheMonth));
+					}
+					finalRRule.append(monthlyReccurencesString);
+					finalRRule.append(";");
+					break;
+			} // end of switch
+ 		}
+		// Handle interval.
+		if (interval > -1) {
+			finalRRule.append("INTERVAL=" + String.valueOf(interval));
+			finalRRule.append(";");
+		}
+		// Handle end.
+		if (endDictionary.containsKey(until)) {
+			// Dictionary can contain only one of the keys,
+			// so what's left is a specific date end rule.
+			finalRRule.append("UNTIL=" + String.valueOf(endDictionary.get(until)));
+			finalRRule.append(";");
+		} else if (endDictionary.containsKey(count)) {
+				// End rule is with occurrence count.
+				finalRRule.append("COUNT=" + String.valueOf(endDictionary.get(count)));
+				finalRRule.append(";");
+			}
+
+		return finalRRule.toString();
+	}
+
+	// Constructor for getting a recurrence rule from the Calendar.
+	public RecurrenceRuleProxy(String nativeRRule, int calendarID, Date begin) {
+		// Only create a proxy if we have a valid recurrence rule.
+		if (nativeRRule != null && !nativeRRule.equals("")) {
+			eventBegin = begin;
+			parseRecurrenceRule(nativeRRule, calendarID);
+		}
+		calculateDaysOfTheMonth();
+		calculateDaysOfTheWeek();
+		calculateDaysOfTheYear();
+		calculateEnd();
+		calculateInterval();
+		calculateMonthsOfTheYear();
+		calculateWeeksOfTheYear();
+	}
+
+	//region Methods for converting native to Kroll values
+	private void calculateDaysOfTheMonth(){
+		String days = matchExpression(".*(BYMONTHDAY=[0-9]*).*", 11);
+		if (days != null && frequency == CalendarModule.RECURRENCEFREQUENCY_MONTHLY) {
+			daysOfTheMonth = new int[1];
+			daysOfTheMonth[0] = Integer.valueOf(days);
+		} else {
+			daysOfTheMonth = new int[]{};
+		}
+	}
+
+	private void calculateDaysOfTheWeek() {
+		// Create a dictionary in the context of a month.
+		if (frequency == CalendarModule.RECURRENCEFREQUENCY_MONTHLY) {
```

Comment:
these two if clauses are exclusive. I think it may make more sense to calculate the necessary fields grouped by the frequency value, rather than to calculate grouped by specific fields.

Basically, initialize days of the week, days of the year, months of the year, etc with the default empty array value (or whatever empty value you decide to use based on Optional/Integer usage); and then override to the real value if the frequency is of a given type in a method that just handles that frequency type.